client/state/data-layer/wpcom/sites/scan/index.js

```diff
@@ -40,17 +40,21 @@ export const formatScanThreat = ( threat ) => ( {
  * @param {object} scanState Raw Scan state object from Scan endpoint
  * @returns {object} Processed Scan state
  */
-const formatScanStateRawResponse = ( { state, threats, credentials, most_recent: mostRecent } ) => {
+const formatScanStateRawResponse = ( {
+	state,
+	threats,
+	credentials,
+	most_recent: mostRecent,
+	current,
+	...rest
+} ) => {
 	return {
 		state,
 		threats: threats.map( formatScanThreat ),
 		credentials,
-		mostRecent: mostRecent
-			? {
-					...mostRecent,
-					isInitial: mostRecent.is_initial,
-			  }
-			: null,
+		...( mostRecent ? { ...mostRecent, isInitial: mostRecent.is_initial } : {} ),
+		...( current ? { ...current, isInitial: current.is_initial } : {} ),
```

Comment:
From the looks of this code, it seems to be returning an empty object instead of undefined?

Reply:
It returns an empty object that then get spread into the parent one. It only creates the key if the condition is true. I use it inside components when I don't want to pass a key is some condition isn't true. Like this:
```
<Component { ...( condition ? { key: 100 } : {} ) } />
```

But I'll change it back to most readable version.